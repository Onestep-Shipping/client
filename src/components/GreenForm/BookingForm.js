import React from 'react';
import '../../pages/_user/Booking/Booking.css';
import PropTypes from 'prop-types';
import {InfoRow, Select} from './Helpers.js';

const BookingForm = (props) => {
  const CONTAINER_TYPES = ['20\' Dry', '40\' Dry', '40\'HC Dry'];
  const PAYMENT_TYPES = ['Prepaid', 'Collect'];

  return (
    <form className="booking-form-container" onSubmit={props.action} noValidate>
      <InfoRow name="Commodity" />
      <InfoRow name="HS Code" />
      <Select name="Container Type" options={CONTAINER_TYPES} />
      <InfoRow name="Quantity" />
      <Select name="Payment Term" options={PAYMENT_TYPES} />
      <button className="result-button">Book</button>
    </form>
  );
};

export default BookingForm;

BookingForm.propTypes = {
  action: PropTypes.func,
};
