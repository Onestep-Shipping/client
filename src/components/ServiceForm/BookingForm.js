import React, {useState} from 'react';
import '../../pages/_user/Booking/Booking.css';
import { 
  CONTAINER_HEADERS, CONTAINER_TYPES, PAYMENT_TYPES, AUTO_FILLING 
} from '../../data/ServiceFormData.js';
import PropTypes from 'prop-types';
import {InfoRow, Select} from './Helpers.js';

const MAX_ROW = 5;

const BookingForm = (props) => {
  const [row, setRow] = useState(1);

  const addContainer = (e) => {
    e.preventDefault();
    const newVal = row + 1;
    if (newVal <= MAX_ROW) {
      setRow(row + 1);
    } else {
      alert("You reached the limit.")
    }
  }

  return (
    <form className="booking-form-container" onSubmit={props.action} noValidate>
      <InfoRow name="Commodity" />
      <InfoRow name="HS Code" />
      <div className="schedule-result-header-row">
        {CONTAINER_HEADERS.map((header, ind) => 
          <div key={ind}>
            <text className="info-label-special">{header}</text>
          </div>
        )}
      </div>
      {Array(row).fill().map((booking, ind) => 
        <div className='instruction-result-row' key={ind}>
          <input
            type="text"
            name="quantity"
            className="booking-form-input-start"
            placeholder="i.e. 420010"
            required/>
          <select className="booking-form-input">
            {CONTAINER_TYPES.map((type, ind) => <option value={type} key={ind}>{type}</option>)}
          </select>
        </div>
      )}
      <button className="add-button" onClick={addContainer}>
        Add Container
      </button>
      <Select name="Payment Term" options={PAYMENT_TYPES} />
      <Select name="CAED/AES filing by OneStep?" options={AUTO_FILLING} />
      <button className="result-button">Book</button>
    </form>
  );
};

export default BookingForm;

BookingForm.propTypes = {
  action: PropTypes.func,
};
