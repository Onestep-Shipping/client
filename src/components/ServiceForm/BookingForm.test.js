import BookingForm from './BookingForm';
import React from 'react';
import { shallow } from 'enzyme';

const mockSubmit = jest.fn();

let wrapper = shallow(
  <BookingForm action={mockSubmit} />
);

describe('BookingForm', () => {
  it('should render the BookingForm Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});