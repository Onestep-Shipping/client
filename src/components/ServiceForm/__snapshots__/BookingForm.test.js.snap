// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingForm should render the BookingForm Component correctly 1`] = `
<form
  className="booking-form-container"
  noValidate={true}
  onSubmit={[MockFunction]}
>
  <InfoRow
    name="Commodity"
  />
  <InfoRow
    camelizeName="hsCode"
    name="HS Code"
  />
  <div
    className="schedule-result-header-row"
  >
    <div
      key="0"
    >
      <span
        className="info-label-special"
      >
        Quantity
      </span>
    </div>
    <div
      key="1"
    >
      <span
        className="info-label-special"
      >
        Container Type
      </span>
    </div>
  </div>
  <div
    className="instruction-result-row"
    key="0"
  >
    <input
      className="booking-form-input-start"
      name="quantity"
      placeholder="i.e. 420010"
      required={true}
      type="number"
    />
    <select
      className="booking-form-input"
      name="containerType"
    >
      <option
        key="0"
        value="20' Dry"
      >
        20' Dry
      </option>
      <option
        key="1"
        value="40' Dry"
      >
        40' Dry
      </option>
      <option
        key="2"
        value="40'HC Dry"
      >
        40'HC Dry
      </option>
    </select>
  </div>
  <button
    className="add-button"
    onClick={[Function]}
  >
    Add Container
  </button>
  <Select
    camelizeName="paymentTerm"
    name="Payment Term"
    options={
      Array [
        "Prepaid",
        "Collect",
      ]
    }
  />
  <Select
    camelizeName="autoFilling"
    name="CAED/AES filing by OneStep?"
    options={
      Array [
        "Yes",
        "No",
      ]
    }
  />
  <button
    className="result-button"
  >
    Book
  </button>
</form>
`;
